src/Integration/Binding/BindingWorkflow.cs

```diff
@@ -1,395 +1,372 @@
-            {
-                projectRuleSetPath = this.projectRuleSetWriter.WriteProjectLevelRuleSet(projectFullPath, configurationName, solutionRuleSetPath, currentRuleSet);
-            }
-
-            return projectRuleSetPath;
-        }
-
-        internal /*for testing purposes*/ void EmitBindingCompleteMessage(IProgressStepExecutionEvents notifications)
-        {
-            var message = this.AllNuGetPackagesInstalled
-                ? Strings.FinishedSolutionBindingWorkflowSuccessful
-                : Strings.FinishedSolutionBindingWorkflowNotAllPackagesInstalled;
-            notifications.ProgressChanged(message, double.NaN);
-        }
-
-        #endregion
-    }
-}
+//-----------------------------------------------------------------------
+// <copyright file="BindingWorkflow.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.Alm.Authentication;
+using Microsoft.VisualStudio.CodeAnalysis.RuleSets;
+using SonarLint.VisualStudio.Integration.Persistence;
+using SonarLint.VisualStudio.Integration.Progress;
+using SonarLint.VisualStudio.Integration.Resources;
+using SonarLint.VisualStudio.Integration.Service;
+using SonarLint.VisualStudio.Integration.Service.DataModel;
+using SonarLint.VisualStudio.Integration.TeamExplorer;
+using SonarLint.VisualStudio.Progress.Controller;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Threading;
+
+namespace SonarLint.VisualStudio.Integration.Binding
+{
+    /// <summary>
+    /// Workflow execution for the bind command
+    /// </summary>
+    internal class BindingWorkflow
+    {
+        private readonly BindCommand owner;
+        private readonly ProjectInformation project;
+        private readonly IProjectSystemHelper projectSystemHelper;
+        private readonly IBindingOperation solutionBindingOperation;
+        private readonly ISolutionRuleStore solutionRuleStore;
+
+        internal readonly Dictionary<string, RuleSetGroup> LanguageToGroupMapping = new Dictionary<string, RuleSetGroup>
+        {
+            {SonarQubeServiceWrapper.CSharpLanguage, RuleSetGroup.CSharp },
+            {SonarQubeServiceWrapper.VBLanguage, RuleSetGroup.VB }
+        };
+
+        public BindingWorkflow(BindCommand owner, ProjectInformation project)
+            : this(owner, project, null, null, null)
+        {
+
+        }
+
+        internal /*for testing purposes*/ BindingWorkflow(BindCommand owner, ProjectInformation project, IProjectSystemHelper projectSystemHelper, IBindingOperation solutionBindingOperation, ISolutionRuleStore solutionRuleStore)

```

Comment:
This ctor is only used in two places:
1. chained with the public ctor above, and
2. in one test helper method, which passes in `new ConfigurableBindingOperation()` and `new ConfigurableSolutionRuleStore()`.
I think really I'm struggling to see why the SolutionBindingOperator needs to implement IBindingOperator as well as ProjectBindingOperator. We don't ever deal with the Solution- and Project- implementors as a whole. Perhaps IBindingOperator should be IProjectBindingOperator. Just because they might share the same methods doesn't mean that they're conceptually the same thing.